pkg/driver/bigquery/bigquery.go

```diff
@@ -66,30 +51,89 @@ func (drv *Driver) CreateDatabase() error {
 		return err
 	}
 
-	if !exists {
-		err := drv.client.Dataset(drv.datasetID).Create(*drv.context, &bigquery.DatasetMetadata{})
+	if exists {
+		return nil
+	}
+
+	ctx := context.Background()
+	con, err := db.Conn(ctx)
+	if err != nil {
+		return err
+	}
+	defer con.Close()
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+
+	err = con.Raw(func(driverConn any) error {
+		defer wg.Done()
+
+		client := getClient(driverConn)
+		dataset := getDataset(driverConn)
+		err := client.Dataset(dataset).Create(ctx, &bigquery.DatasetMetadata{})
 		if err != nil {
 			return err
 		}
+		return nil
+	})
+
+	wg.Wait()
```

Comment:
My main point of concern was `con.Raw()` callback because from my limited knowledge of Golang, callbacks would not be sync.
![image](https://github.com/amacneil/dbmate/assets/6401874/b10d9fe4-98c5-4fba-8465-f0f39e69dcf9)

Reply:
But I did run the test file without `WaitGroup` once or twice and it ran fine. But when debugging, sometimes I noticed, the flow would go directly to the last statement of the function and then go into the callback

Reply:
removed all `WaitGroup`s